src/test/java/org/sagebionetworks/bridge/services/TemplateServiceTest.java

```diff
@@ -0,0 +1,575 @@
+        assertSame(result, template);
+        assertSame(result.getCriteria(), template.getCriteria());
+        assertEquals(result.getCriteria().getKey(), "template:"+GUID1);
+    }
+    
+    @Test(expectedExceptions = BadRequestException.class)
+    public void getTemplateNoGuid() {
+        service.getTemplate(TEST_STUDY, null);
+    }
+
+    @Test(expectedExceptions = EntityNotFoundException.class)
+    public void getTemplateNotFound() {
+        when(mockTemplateDao.getTemplate(TEST_STUDY, GUID1)).thenReturn(Optional.empty());
+        
+        service.getTemplate(TEST_STUDY, GUID1);
+    }
+    
+    @Test
+    public void createTemplate() {
+        doAnswer(answer -> {
+            return answer.getArgument(0);
+        }).when(mockCriteriaDao).createOrUpdateCriteria(any());
+        doAnswer(answer -> {
+            Template captured = answer.getArgument(0);
+            captured.setVersion(10);
+            return null;
+        }).when(mockTemplateDao).createTemplate(any());
+        
+        Criteria criteria = Criteria.create();
+        criteria.setAllOfGroups(ImmutableSet.of("group1", "group2"));
+        
+        Template template = Template.create();
+        template.setName("Test name");
+        template.setTemplateType(EMAIL_RESET_PASSWORD);
+        template.setDeleted(true);
+        template.setVersion(3);
+        template.setCriteria(criteria);
+        
+        GuidVersionHolder holder = service.createTemplate(TEST_STUDY, template);
+        assertEquals(holder.getGuid(), GUID1);
+        assertEquals(holder.getVersion(), new Long(10));
+        
+        assertEquals(template.getStudyId(), TEST_STUDY_IDENTIFIER);
+        assertFalse(template.isDeleted());
+        assertEquals(template.getVersion(), 10);
+        assertEquals(template.getGuid(), GUID1);
+        assertEquals(template.getCreatedOn(), TIMESTAMP);
+        assertEquals(template.getModifiedOn(), TIMESTAMP);
+        assertEquals(template.getCriteria().getKey(), "template:"+GUID1);
+        
+        verify(mockCriteriaDao).createOrUpdateCriteria(criteria);
+        verify(mockTemplateDao).createTemplate(template);
+    }
+        
+    @Test
+    public void createTemplateDefaultsCriteria() {
+        Template template = Template.create();
+        template.setName("Test");
+        template.setTemplateType(EMAIL_RESET_PASSWORD);
+        
+        service.createTemplate(TEST_STUDY, template);
+        
+        verify(mockCriteriaDao).createOrUpdateCriteria(any(Criteria.class));
+    }
+    
+    @Test(expectedExceptions = InvalidEntityException.class)
+    public void createTemplateInvalid() {
+        service.createTemplate(TEST_STUDY, Template.create());
+    }
+    
+    @Test
+    public void updateTemplate() {
+        doAnswer(answer -> {
+            Template captured = answer.getArgument(0);
+            captured.setVersion(10);
+            captured.setModifiedOn(TIMESTAMP);
```

Comment:
TemplateService.updateTemplate() already sets the modifiedOn, and the dao doesn't. Wouldn't setting the template here actually be incorrect?

Reply:
You're right, removed this, test is unaffected